gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -822,6 +822,54 @@ public static AddressFilter notEquals(AddressField field, String value) {
   }
 
   /**
+   * Class for filtering snapshot lists.
+   */
+  class SnapshotFilter extends ListFilter {
+
+    private static final long serialVersionUID = 8757711630092406747L;
+
+    SnapshotFilter(SnapshotField field, ComparisonOperator operator, Object value) {
+      super(field.selector(), operator, value);
+    }
+
+    /**
+     * Returns an equality filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static SnapshotFilter equals(SnapshotField field, String value) {
```

Comment:
I think finding method that resemble in name other common methods (such as equals hashCode,...) is a nice feature as it helps with undetected typos (and I think that is a better help than `@Override` which can be easily forgotten,
especially because that is typically auto inserted by IDEs but not in a case of a typo).

I would even take that check further and compare any method name with all its parent class and interface hierarchy for similar typos.

However, I think this check should only be triggered when the signature is the same (or close). In this case being static and with 2 arguments I think that check is plain wrong.